compute_endpoint/globus_compute_endpoint/endpoint/endpoint.py

```diff
@@ -655,62 +656,80 @@ def stop_endpoint(
 
-        if not endpoint_dir.exists():
-            log.warning(f"Endpoint <{ep_name}> does not exist")
+        if ep_dir:
+            ep_name = str(ep_dir)
+        elif ep_uuid:
+            ep_name = ep_uuid
+        else:
+            log.warning("Name or UUID is needed to delete an Endpoint")
             exit(-1)
 
-        endpoint_id = Endpoint.get_endpoint_id(endpoint_dir)
-        if endpoint_id is None:
-            log.warning(
-                f"Configuration for endpoint <{ep_name}> could not be found, it "
-                "might not have been initialized locally"
-            )
-            if not force:
+        # If we have the UUID, do the online status check/deletion first
+        if ep_uuid:
+            try:
+                uuid.UUID(ep_uuid)
+            except (ValueError, TypeError):
+                log.warning(f"{ep_uuid} is not a valid endpoint UUID")
                 exit(-1)
 
-        # Delete endpoint from web service
-        try:
-            fx_client = Endpoint.get_funcx_client(endpoint_config)
-            fx_client.delete_endpoint(endpoint_id)
-            log.info(f"Endpoint <{ep_name}> has been deleted from the web service")
-        except AuthAPIError:
-            # Send to the exception handler
-            raise
-        except GlobusAPIError as e:
-            log.warning(
-                f"Endpoint <{ep_name}> could not be deleted from the web service"
-                f"  [{e.text}]"
-            )
-            if not force:
-                log.critical("Exiting without deleting the endpoint")
-                exit(os.EX_UNAVAILABLE)
-        except NetworkError as e:
-            log.warning(
-                f"Endpoint <{ep_name}> could not be deleted from the web service"
-                f"  [{e}]"
-            )
-            if not force:
-                msg = (
-                    "globus-compute-endpoint is unable to reach the Globus Compute "
-                    "service due to a network error.\n"
-                    "Please ensure the Globus Compute service address is reachable, "
-                    "then attempt to delete the endpoint again."
+            try:
+                gc_client = Endpoint.get_funcx_client(ep_config)
+                status = gc_client.get_endpoint_status(ep_uuid)
+                if status.get("status") == "online":
+                    if not force:
+                        log.warning(
+                            f"Endpoint {ep_uuid} is currently running.  To proceed "
+                            "with deletion, first suspend it with the command "
+                            "`globus-compute-endpoint stop --remote {ep_uuid}` or "
+                            "ignore the current status with `delete --force`"
+                        )
+                        exit(-1)
+                    if ep_dir:
+                        # Stop endpoint to handle process cleanup
+                        Endpoint.stop_endpoint(ep_dir, ep_config, remote=False)
```

Comment:
If we are going to check the status of the endpoint before deleting it from the web service, then we should attempt to stop it beforehand.

Reply:
This was intentional - a reminder that the endpoint is live, adding a second check in addition to the --yes in case he/she didn't realize it's active.

Reply:
That's fine, but then we should adjust the warning message in the event that the web service reports the endpoint online. The current message recommends that the user stops the endpoint with `globus-compute-endpoint stop --remote {ep_uuid}`, which implies that the endpoint is not running locally.

Another route is to check the status of the local endpoint process with `Endpoint.check_pidfile()`, as we do in `Endpoint.stop_endpoint()`, then either raise a warning or call `Endpoint.stop_endpoint()`, then proceed with checking the status from the web service.

Reply:
message shortened to be less specific